source/dagger/gameplay/ping_pong/pingpong_ball.cpp

```diff
@@ -55,6 +56,16 @@ void PingPongBallSystem::Run()
                 if (std::abs(collisionSides.x) > 0)
                 {
                     ball.speed.x *= -1;
+					if (Engine::Registry().view<ControllerMapping>().contains(col.colidedWith)) { // If the entity the ball colided with has controls mapped
+																								  // it is a player and should stop moving for a bit of time
+						auto &player = Engine::Registry().view<ControllerMapping>().get(col.colidedWith);
+						
+						if (!player.frozen) { // Player might already be frozen due to a boost, in that case dont freeze him again
+							player.frozen = true;
+							player.frozenFor = 3.0f; // Freeze for 3 seconds
```

Comment:
you can put 3 sec as const to ControllerMapping class;

static constexpr float s_frozenMaxTime = 3.f; // or something like that